File: app/src/main/java/com/kingkharnivore/skillz/ui/screen/paths/suggested/SuggestedRoutesCatalog.kt
package com.kingkharnivore.skillz.ui.screen.paths.suggested

import com.kingkharnivore.skillz.model.ui.SuggestedRouteStepUiModel
import com.kingkharnivore.skillz.model.ui.SuggestedRouteUiModel

object SuggestedRoutesCatalog {

    val routes: List<SuggestedRouteUiModel> = listOf(
        SuggestedRouteUiModel(
            id = "deep_work_launch",
            title = "Deep Work Launch",
            subtitle = "Start serious work without drifting.",
            category = "Focus",
            approxMinutes = 75,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Clear the Desk",
                    tagName = "Focus",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Define the Target",
                    tagName = "Planning",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Deep Work Block",
                    tagName = "Work",
                    isSoftMode = false,
                    targetMinutes = 45,
                    launchWithSurge = true
                ),
                SuggestedRouteStepUiModel(
                    title = "Capture Next Action",
                    tagName = "Planning",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "learn_and_lock_in",
            title = "Learn and Lock In",
            subtitle = "Turn study into retained knowledge.",
            category = "Learning",
            approxMinutes = 90,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Read or Watch Lesson",
                    tagName = "Learning",
                    isSoftMode = false,
                    targetMinutes = 30,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Practice Actively",
                    tagName = "Practice",
                    isSoftMode = false,
                    targetMinutes = 30,
                    launchWithSurge = true
                ),
                SuggestedRouteStepUiModel(
                    title = "Write Notes",
                    tagName = "Notes",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Recall From Memory",
                    tagName = "Learning",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "ship_one_thing",
            title = "Ship One Thing",
            subtitle = "Finish one tangible piece before momentum fades.",
            category = "Build",
            approxMinutes = 105,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Pick the Smallest Shippable Piece",
                    tagName = "Planning",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Build the Core",
                    tagName = "Build",
                    isSoftMode = false,
                    targetMinutes = 60,
                    launchWithSurge = true
                ),
                SuggestedRouteStepUiModel(
                    title = "Polish the Edge",
                    tagName = "Craft",
                    isSoftMode = false,
                    targetMinutes = 20,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Publish or Commit",
                    tagName = "Ship",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "reset_room_reset_mind",
            title = "Reset the Room, Reset the Mind",
            subtitle = "Recover control of your space and attention.",
            category = "Reset",
            approxMinutes = 45,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Clear Visible Clutter",
                    tagName = "Home",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Reset One Surface",
                    tagName = "Home",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Prep the Next Task",
                    tagName = "Planning",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Soft Reflection",
                    tagName = "Reflection",
                    isSoftMode = true,
                    targetMinutes = 10,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "evening_closeout",
            title = "Evening Closeout",
            subtitle = "End the day with fewer open loops.",
            category = "Evening",
            approxMinutes = 35,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Review What Moved",
                    tagName = "Reflection",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Capture Unfinished Loops",
                    tagName = "Notes",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Choose Tomorrow’s First Flow",
                    tagName = "Planning",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Wind Down",
                    tagName = "Wellness",
                    isSoftMode = true,
                    targetMinutes = 5,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "creative_forge",
            title = "Creative Forge",
            subtitle = "Shape raw sparks into something you can return to.",
            category = "Creative",
            approxMinutes = 90,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Gather Sparks",
                    tagName = "Creative",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Make the Rough Version",
                    tagName = "Creative",
                    isSoftMode = false,
                    targetMinutes = 45,
                    launchWithSurge = true
                ),
                SuggestedRouteStepUiModel(
                    title = "Refine One Section",
                    tagName = "Craft",
                    isSoftMode = false,
                    targetMinutes = 20,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Save the Next Thread",
                    tagName = "Notes",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = false
                )
            )
        ),
        SuggestedRouteUiModel(
            id = "body_before_battle",
            title = "Body Before Battle",
            subtitle = "Prepare your body before intense focus.",
            category = "Prep",
            approxMinutes = 35,
            steps = listOf(
                SuggestedRouteStepUiModel(
                    title = "Walk or Warm Up",
                    tagName = "Fitness",
                    isSoftMode = false,
                    targetMinutes = 15,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Hydrate",
                    tagName = "Wellness",
                    isSoftMode = false,
                    targetMinutes = 5,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Quick Reset",
                    tagName = "Focus",
                    isSoftMode = false,
                    targetMinutes = 5,
                    launchWithSurge = false
                ),
                SuggestedRouteStepUiModel(
                    title = "Begin Focus Block",
                    tagName = "Work",
                    isSoftMode = false,
                    targetMinutes = 10,
                    launchWithSurge = true
                )
            )
        )
    )

    fun getById(id: String): SuggestedRouteUiModel? =
        routes.firstOrNull { it.id == id }
}

File: ios/Scyra/Scyra/Domain/Horizon/SuggestedRoutesCatalog.swift
import Foundation

enum SuggestedRoutesCatalog {
    static let routes: [SuggestedRoute] = [
        route(
            "deep_work_launch", "Deep Work Launch", "Start serious work without drifting.", "Focus",
            [("Clear the Desk", "Focus", 10, false), ("Define the Target", "Planning", 10, false),
             ("Deep Work Block", "Work", 45, true), ("Capture Next Action", "Planning", 10, false)]
        ),
        route(
            "learn_and_lock_in", "Learn and Lock In", "Turn study into retained knowledge.", "Learning",
            [("Read or Watch Lesson", "Learning", 30, false), ("Practice Actively", "Practice", 30, true),
             ("Write Notes", "Notes", 15, false), ("Recall From Memory", "Learning", 15, false)]
        ),
        route(
            "ship_one_thing", "Ship One Thing", "Finish one tangible piece before momentum fades.", "Build",
            [("Pick the Smallest Shippable Piece", "Planning", 15, false), ("Build the Core", "Build", 60, true),
             ("Polish the Edge", "Craft", 20, false), ("Publish or Commit", "Ship", 10, false)]
        ),
        route(
            "reset_room_reset_mind", "Reset the Room, Reset the Mind",
            "Recover control of your space and attention.", "Reset",
            [("Clear Visible Clutter", "Home", 15, false), ("Reset One Surface", "Home", 10, false),
             ("Prep the Next Task", "Planning", 10, false), ("Soft Reflection", "Reflection", 10, false)]
        ),
        route(
            "evening_closeout", "Evening Closeout", "End the day with fewer open loops.", "Evening",
            [("Review What Moved", "Reflection", 10, false), ("Capture Unfinished Loops", "Notes", 10, false),
             ("Choose Tomorrow’s First Flow", "Planning", 10, false), ("Wind Down", "Wellness", 5, false)]
        ),
        route(
            "creative_forge", "Creative Forge", "Shape raw sparks into something you can return to.", "Creative",
            [("Gather Sparks", "Creative", 15, false), ("Make the Rough Version", "Creative", 45, true),
             ("Refine One Section", "Craft", 20, false), ("Save the Next Thread", "Notes", 10, false)]
        ),
        route(
            "body_before_battle", "Body Before Battle", "Prepare your body before intense focus.", "Prep",
            [("Walk or Warm Up", "Fitness", 15, false), ("Hydrate", "Wellness", 5, false),
             ("Quick Reset", "Focus", 5, false), ("Begin Focus Block", "Work", 10, true)]
        )
    ]

    private static func route(
        _ id: String,
        _ title: String,
        _ subtitle: String,
        _ category: String,
        _ steps: [(String, String, Int, Bool)]
    ) -> SuggestedRoute {
        SuggestedRoute(
            id: id,
            title: title,
            subtitle: subtitle,
            category: category,
            approximateMinutes: steps.reduce(0) { $0 + $1.2 },
            steps: steps.map {
                SuggestedRouteStep(
                    title: $0.0,
                    journeyName: $0.1,
                    targetMinutes: $0.2,
                    launchWithSurge: $0.3
                )
            }
        )
    }
}
